open-grant-proposals/Filecoin Financing Plan.md

```diff
@@ -0,0 +1,122 @@
+To secure the benefit of the investors, the financing plan itself shows investors the credit information of the financier. At the same time, the logic of freezing financing fund and income automatic repayment of the financing plan will be implemented in the lotus-miner, and to freeze financing fund will guarantee that the financing fund can only be used to increase the mining pool power. The Automatic income repayment will guarantee that the investors' income is not subject to the individual intentions of the financier. 
+
+## Value
+
+Please describe in more detail why this proposal is valuable for the Filecoin ecosystem. Answer the following questions:
+- What are the benefits to getting this right?
+A: The project can spread the pressure of high cost to join Filecoin, to reduce the risk for individuals to participate, to lower the cost threshold for Filecoin participation, and to attract more users to get involved.
+- What are the risks if you don't get it right?
+A: If we didn't get it right, the investors may lost their capital.
+- What are the risks that will make executing on this project difficult?
+A: The credit rating of financiers is not comprehansive or accurate enough, and that makes investors afraid to invest. 
+
+## Deliverables
+
+1. deliver the lotus-miner implementation code for financing plan
+2. deliver the web browing portal (financing plan explorer) code for financing plan and access services.
+
+## Technical Breakdown
+
+Filecoin financing plan detailed process:
+1. Add two control address types for miner, which means on the basis of the worker/owner address, we add the financing address and the repayment freeze address.
+2. A sealing sector first determines whether the financing address has some balance, If the address has, the sector will use the balance; if not, the sealing sector will follow the original logic to process.
+3. A financing plan includes features below:
+Financing address
+Financing start time
+Financing end time
+Financing total amount (At the financing end time, if financing total amount has not yet met, the financing plan will be canceled automatically.)
+Repayment total amount
+Repayment start time
+Repayment end time
+Repayment period
+Repayment period option (3 options: month, week, day)
+4. During the repayment time, the miner balance is frozen to pay for the current repayment. When the added-up balance meets the repayment limit, the balance will be automatically transferred to the repayment freeze address.
+5. Only when the balance in the freeze repayment address has met the repayment amount, the miner’s balance can be withdrew and used. 
+6. When repayment time is about to end, repayment freeze address will automatically transfer the scheduled repayment amount into the financing investment address.
```

Comment:
In the current network there is a high technical bar that needs to be cleared to modify the state machine.  As it stands I don't think this proposal clears it.  It's possible I'm wrong but I will need a more detailed spec of the actors changes to be convinced.

Reply:
However I think I see an acceptable way to get what you want so I'm including a counter proposal.